src/send/usePrepareSendTransactions.ts

```diff
@@ -0,0 +1,108 @@
+import { useState } from 'react'
+import {
+  getMaxGasCost,
+  PreparedTransactionsResult,
+  prepareERC20TransferTransaction,
+} from 'src/viem/prepareTransactions'
+import { TokenBalance, tokenBalanceHasAddress } from 'src/tokens/slice'
+import BigNumber from 'bignumber.js'
+import { useAsyncCallback } from 'react-async-hook'
+import { tokenSupportsComments } from 'src/tokens/utils'
+import Logger from 'src/utils/Logger'
+import { tokenAmountInSmallestUnit } from 'src/tokens/saga'
+
+const TAG = 'src/send/usePrepareSendTransactions'
+
+// just exported for testing
+export function _getOnSuccessCallback({
+  setFeeCurrency,
+  setPrepareTransactionsResult,
+  setFeeAmount,
+}: {
+  setFeeCurrency: (token: TokenBalance | undefined) => void
+  setPrepareTransactionsResult: (result: PreparedTransactionsResult | undefined) => void
+  setFeeAmount: (amount: BigNumber | undefined) => void
+}) {
+  return (result: PreparedTransactionsResult | undefined) => {
+    setPrepareTransactionsResult(result)
+    if (result?.type === 'possible') {
+      setFeeCurrency(result.feeCurrency)
+      setFeeAmount(
+        getMaxGasCost(result.transactions).dividedBy(
+          new BigNumber(10).exponentiatedBy(result.feeCurrency.decimals)
+        )
+      )
+    } else {
+      setFeeCurrency(undefined)
+      setFeeAmount(undefined)
```

Comment:
these are for the two warn states right? Should we still set these so a fee + warning is displayed instead of the placeholder?

Reply:
if there's not enough balance for gas, it's unclear which fee currency a gas cost should be shown for. Extra, potentially cumbersome logic would be needed in prepareTransactions to handle that.

for the 'need-decrease-spend-amount-for-gas' case, sure